specs/agents/configuration.md

```diff
@@ -112,3 +112,17 @@ Failure to process one config attribute should not affect processing of others.
 #### Feature flag
 
 Agents should implement a [configuration option](https://docs.google.com/spreadsheets/d/1JJjZotapacA3FkHc2sv_0wiChILi3uKnkwLTjtBmxwU), (`CENTRAL_CONFIG`) which lets users disable the central configuration polling.
+
+### Zero-configuration support
+
+To decrease initial setup costs per application,
+APM agents should not require any configuration to be set
+in order to start sending data to a local APM Server.
+After the initial onboarding, users can still customize settings
+if the defaults aren't fitting
+but that should not stand in the way of the initial onboarding process.
+
+Agents should be able to auto-detect the APM Server endpoint
+by just defaulting to `http://localhost:8200/`.
+Furthermore, if it's possible, agents should detect a sensible defaults for `service.name`
+as well as the `service.version`.
```

Comment:
```suggestion
By default, agents should send to the APM Server at `http://localhost:8200/`.
If possible, agents should detect sensible defaults for `service.name` and `service.version`.
Even if detecting these values is not possible, a default value should be provided for
frictionless onboarding.
```

This takes @trentm's suggestions into account as well, so if you like this suggestion please commit this one instead of Trent's.

Reply:
I agree with your suggestion but I think it might be worth adding that having default for `service.name` is more important than `service.version`. Maybe we can even go further and say that default (to fall back on when detection is not available) is only necessary for `service.name` and `service.version` can be just omitted. WDYT?

Reply:
Yes, that sounds good.

Reply:
Agreed.